wagtail/admin/panels.py

```diff
@@ -527,9 +529,23 @@ def visible_children_with_identifiers(self):
                 if child.is_shown()
             ]
 
-        def is_shown(self):
+        def any_children_shown(self):
+            """
+            Helper function that calls `is_shown` on any child panels the group
+            contains.
+            """
             return any(child.is_shown() for child in self.children)
 
+        def is_shown(self):
+            if self.panel.permission:
+                # If a permission is set and passes, just check the children.
+                if self.request.user.has_perm(self.panel.permission):
+                    return self.any_children_shown()
+                return False
+
+            # If no permissions, just check regular.
+            return self.any_children_shown()
```

Comment:
Let's try to avoid adding this `any_children_shown`. It adds a bit of indirection and the logic can read pretty well with a small rework.

Rough version below
```python
        def is_shown(self):
            """
            Check permissions on the panel group overall then check if any children
            are shown.
            """

            if self.panel.permission:
                if not self.request.user.has_perm(self.panel.permission):
                    return False

            return any(child.is_shown() for child in self.children)
```